third_party/libpg_query/grammar/statements/alter_table.y

```diff
@@ -298,15 +298,16 @@ alter_table_cmd:
 				}
 			/*
 			 * ALTER TABLE <name> ALTER [COLUMN] <colname> [SET DATA] TYPE <typename>
-			 *		[ USING <expression> ]
+			 *		[ USING <expression> ] [RESTRICT|CASCADE]
 			 */
-			| ALTER opt_column ColId opt_set_data TYPE_P Typename opt_collate_clause alter_using
+			| ALTER opt_column ColId opt_set_data TYPE_P Typename opt_collate_clause alter_using opt_drop_behavior
```

Comment:
It looks like `CASCADE` is not supported with `ALTER TYPE` in Postgres, and intuitively it is not exactly clear to me what `CASCADE` would do with `ALTER TYPE` either. Perhaps remove this functionality?

Reply:
The idea was that when we change the type of a column that is a dependency of a generated column - this breaks the generated column

So we would only allow this change to go through if CASCADE was provided
That way it's clear to the user that this statement will delete the generated column

But it probably isn't worth the cost of adding CASCADE and RESTRICT to the reserved keywords list